app/engines/i14y_formatted_query.rb

```diff
@@ -11,12 +11,18 @@ def delimiters
   end
 
   def query_with_sites(user_query)
-    remaining_chars = QUERY_STRING_ALLOCATION - user_query.length
-    domains = fill_included_domains_to_remainder(remaining_chars)
-    excluded =  fill_excluded_domains_to_remainder(remaining_chars - domains.length)
+    site_no_minus_site = user_query.include?('site:') &&
+                         !user_query.include?('-site:')
+    current_chars = QUERY_STRING_ALLOCATION - user_query.length
+    domains = site_no_minus_site ? '' : fill_included_domains_to_remainder(current_chars)
+    excluded = fill_excluded_domains_to_remainder(current_chars - domains.length)
+    get_sites(user_query, excluded, domains)
+  end
+
+  def get_sites(user_query, excluded, domains)
     sites = [user_query]
-    sites << excluded unless excluded.blank?
-    sites << domains unless domains.blank?
+    sites << excluded if excluded.present?
+    sites << domains if domains.present?
     sites.join(' ')
   end
 end
```

Comment:
Rubocop reported that this method complexity was high. The method was doing too many things and too many lines. I broke it into two methods to make Rubocop happy.